sw/device/lib/crypto/drivers/entropy.c

```diff
@@ -422,12 +403,14 @@ static status_t csrng_send_app_cmd(uint32_t base_address,
   abs_mmio_write32(cmd_reg_addr, reg);
 
   for (size_t i = 0; i < cmd_len; ++i) {
-    // If the command is issued to the SW register of the EDN, the reg ready
-    // bit needs to be polled before writing each word of additional data.
-    if (cmd_type == kEntropyCsrngSendAppCmdTypeEdnSw) {
+    // Before writing each word of additional data, the command ready or command
+    // reg ready bit needs to be polled if the command is issued to CSRNG or the
+    // SW register of EDN, respectively.
+    if (cmd_type == kEntropyCsrngSendAppCmdTypeCsrng ||
+        cmd_type == kEntropyCsrngSendAppCmdTypeEdnSw) {
       do {
-        reg = abs_mmio_read32(base_address + EDN_SW_CMD_STS_REG_OFFSET);
-        ready = bitfield_bit32_read(reg, EDN_SW_CMD_STS_CMD_REG_RDY_BIT);
+        reg = abs_mmio_read32(sts_reg_addr);
```

Comment:
Since this code will be consumed at higher integration layers, it makes sense to not block forever. Would it make sense to add a timeout mechanism similart to what was done before with `kCsrngIdleNumTries`?

Reply:
This sounds like a good idea @moidx . I've filed #23179 to track this.